pkg/listeners/listeners.go

```diff
@@ -47,39 +41,31 @@ var Listeners = []Listener{
 	{
 		Path:        "/readyz",
 		Methods:     []string{http.MethodGet},
-		HandlerFunc: http.HandlerFunc(RespondOKHandler),
+		HandlerFunc: RespondOKHandler,
 	},
 	{
 		Path:        "/healthz",
 		Methods:     []string{http.MethodGet},
-		HandlerFunc: http.HandlerFunc(RespondOKHandler),
+		HandlerFunc: RespondOKHandler,
 	},
 	{
 		Path:        "/api/v1/alert",
 		Methods:     []string{http.MethodPost},
-		HandlerFunc: http.HandlerFunc(ProcessAlertHandler),
+		HandlerFunc: ProcessAlertHandler,
 	},
 }
 
-// CreateListener creates a new HTTP Handler from the provided path, HTTP method, and handler function
-func CreateListener(path string, methods []string, handlerFunc http.HandlerFunc) Handler {
-	if config.AppConfig.Verbose {
-		log.Println("enabling endpoint", path, methods)
-	}
-
-	return Handler{
-		Route: func(r *mux.Route) {
-			r.Path(path).Methods(methods...)
-		},
-		Func: handlerFunc,
+// InitRoutes initializes routes from the defined Listeners
+func InitRoutes(router *mux.Router) {
+	for _, listener := range Listeners {
+		if config.AppConfig.Verbose {
+			log.Println("enabling endpoint", listener.Path, listener.Methods)
+		}
+		route := router.NewRoute().HandlerFunc(listener.HandlerFunc).Name(listener.Path)
```

Comment:
setting a "name" here is mostly for unit testing purposes, so that we can verify routes have been added to the router after calling this function.